src/components/AddSample/AddSample.jsx

```diff
@@ -0,0 +1,378 @@
+
+import { useDispatch, useSelector } from 'react-redux';
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+// material ui imports 
+import { makeStyles } from '@material-ui/core/styles';
+import { TextField, 
+  Button, MenuItem,
+  FormHelperText, FormControl, Select, Typography } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  inputs: {
+   margin: theme.spacing(2),
+  },
+  formControl: {
+    margin: theme.spacing(2),
+    minWidth: 180,
+  },
+}));
+
+
+function AddSample() {
+  // material ui
+  const classes = useStyles();
+
+  // set stuff so we can use them
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  // get state from redux store
+  const user = useSelector(store => store.user);
+
+  // local states for input fields
+  const [ingredientName, setName] = useState('');
```

Comment:
Keeping all this stuff in local state will work find for _creating a new sample_, but will cause headaches for you when you move to _edit_ mode.

Once you get there, I recommend moving all of this data to a reducer in Redux state, and removing all this local state entirely. I'm happy to talk through that process with you, once you get there